monkey/monkey_island/cc/next_ui/src/app/api/auth/[...nextauth]/authOptions.ts

```diff
@@ -0,0 +1,154 @@
+            } else if (Date.now() < token.expires_at * 1000 - 200) {
+                return token;
+            } else {
+                try {
+                    let refreshTokenResponseBody = null;
+                    if (token?.access_token) {
+                        const options = {
+                            method: HTTP_METHODS.POST,
+                            headers: {
+                                Accept: 'application/json',
+                                'Content-Type': 'application/json',
+                                'Authentication-Token': token.refresh_token
+                            }
+                        };
+
+                        const refreshTokenResponse = await fetch(
+                            `${process.env.BASE_API}/refresh-authentication-token`,
+                            options
+                        );
+                        refreshTokenResponseBody =
+                            await refreshTokenResponse.json();
+                    }
+
+                    if (!refreshTokenResponseBody) {
+                        throw new Error('RefreshAccessTokenError');
+                    }
+
+                    if (refreshTokenResponseBody?.response?.errors) {
+                        throw new Error(
+                            JSON.stringify(
+                                refreshTokenResponseBody?.response?.errors
+                            )
+                        );
+                    }
+
+                    const statusCode = refreshTokenResponseBody?.meta?.code;
+                    if (statusCode && statusCode !== 200) {
+                        throw new Error('RefreshAccessTokenError');
+                    }
+
+                    const newData = refreshTokenResponseBody?.response?.user;
+
+                    if (!newData?.authentication_token) {
+                        throw new Error('RefreshAccessTokenError');
+                    }
+
+                    const secondsToExpire = Math.floor(
+                        Date.now() / 1000 + newData.token_ttl_sec
+                    );
+
+                    return {
+                        ...token, // Keep the previous token properties
+                        access_token: newData.authentication_token,
+                        expires_at: secondsToExpire,
+                        exp: secondsToExpire,
+                        // Fall back to old refresh token
+                        refresh_token: newData.authentication_token
+                    };
+                } catch (error) {
+                    console.error('Error refreshing access token', error);
+
+                    return null; // clear the session if the refresh token gets an error
+
+                    // The error property will be used client-side to handle the refresh token error
+                    // return {
+                    //     ...token,
+                    //     error: 'RefreshAccessTokenError' as const
+                    // };
```

Comment:
Are we going to do this later?

Reply:
The whole authentication needs a lot of work